File: JavaOPP/scoolmanagement/Payment.java
package JavaOPP.scoolmanagement;

import java.text.DecimalFormat;
import java.util.Date;

public class Payment {
    private int id;
    private Student student;
    private float amount;
    private Date paymentDate;


    public Payment(int id, Student student, float amount, Date paymentDate) {
        this.id = id;
        this.student = student;
        this.amount = amount;
        this.paymentDate = paymentDate;
    }


    public int getId() {
        return this.id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public Student getStudent() {
        return this.student;
    }

    public void setStudent(Student student) {
        this.student = student;
    }

    public float getAmount() {
        return this.amount;
    }

    public void setAmount(float amount) {
        this.amount = amount;
    }

    public Date getPaymentDate() {
        return this.paymentDate;
    }

    public void setPaymentDate(Date paymentDate) {
        this.paymentDate = paymentDate;
    }

    public void printPayment() {
        System.out.println("***** Invoice No. " + this.id + " ******");
        System.out.println("Student: " + this.student.getFullName());
        DecimalFormat amountFormat = new DecimalFormat("0.00");
        System.out.println("Amount: " + amountFormat.format(this.amount));
        System.out.println("Date: " + this.paymentDate.toString());
    }
}

File: JavascriptOPP/schoolmanagement/Payment.js
module.exports = class Payment {
    
    constructor(id, amount, student, paymentDate) {
        this.id = id;
        this.student = student;
        this.amount = amount;
        this.paymentDate = paymentDate;
    }

    getId() {
        return this.id;
    }

    getStudent() {
        return this.student;
    }

    getAmount() {
        return this.amount;
    }

    getPaymentDate() {
        return this.paymentDate;
    }

    printInvoice() {
        console.log("\n");
        console.log("**** Invoice No. " + this.id + "   ******");
        console.log("Estudiante: " + this.student.getFullName());
        console.log("Amount: $" + this.amount.toFixed(2));
        console.log("Payment Date: " + this.paymentDate.toString());
        console.log("\n");
    }

}